### PR TITLE
BIGNUMERIC support for BigQuery dialect

### PR DESCRIPTION
**Pull request by tcqiuyu**

### CODE REVIEW

File: tests/dialects/test_bigquery.py [hunk 1]
```diff
@@ -252,6 +252,12 @@ def test_bigquery(self):
                 "bigquery": "CREATE TABLE db.example_table (x INT64) PARTITION BY x CLUSTER BY x",
             },
         )
+        self.validate_all(
+            "CREATE TABLE db.example_table (x BIGNUMERIC) PARTITION BY x cluster by x",
+            write={
+                "bigquery": "CREATE TABLE db.example_table (x BIGNUMERIC) PARTITION BY x CLUSTER BY x",
+            },
+        )
```
Comment (GeorgeSittas) [inline tests/dialects/test_bigquery.py:260]: Let's use `validate_identity` here instead (note: `cluster by` needs to be converted to uppercase).

File: sqlglot/dialects/bigquery.py [hunk 2]
```diff
@@ -249,6 +250,7 @@ class Generator(generator.Generator):
         TYPE_MAPPING = {
             **generator.Generator.TYPE_MAPPING,  # type: ignore
             exp.DataType.Type.BIGINT: "INT64",
+            exp.DataType.Type.BIGNUMERIC: "BIGNUMERIC",
```
Comment (GeorgeSittas) [inline sqlglot/dialects/bigquery.py:253]: This isn't needed.

**Review by GeorgeSittas — COMMENTED** [review]
Hey, thanks for the PR. Left a couple of comments but looks good otherwise.

Can you please make sure to install the latest dev dependencies & run `make style`? There's no reason to do most of these formatting changes, since they're not required by black.

**Comment by tcqiuyu** [comment]
Thanks. I will check on that. Maybe it is autoformatted by vscode. 😂

File: tests/dialects/test_bigquery.py (hunk 1 above)
Comment (tcqiuyu) [reply tests/dialects/test_bigquery.py:260]: ACK. I will change this.

**Comment by tcqiuyu** [comment]
Finished. Thanks for reviewing!

File: sqlglot/dialects/bigquery.py (hunk 2 above)
Comment (tcqiuyu) [reply sqlglot/dialects/bigquery.py]: ACK

**Review by GeorgeSittas — APPROVED** [review]
LGTM, thanks!

**Comment by GeorgeSittas** [comment]
Closing this in favor of https://github.com/tobymao/sqlglot/commit/53321211a941867589b63237980a9e71fd77f0f2 (see related discussion: https://github.com/tobymao/sqlglot/issues/1381#issuecomment-1513459711).

**Pull request closed without merging**